src/dtl/dyad_dtl_impl.h

```diff
@@ -0,0 +1,64 @@
+#ifndef DYAD_DTL_IMPL_H
+#define DYAD_DTL_IMPL_H
+
+#include "dyad_rc.h"
+#include "dyad_flux_log.h"
+
+#include <flux/core.h>
+#include <jansson.h>
+
+#include "dyad_dtl.h"
+
+#ifdef __cplusplus
+#include <cstdint>
+
+extern "C" {
+#else
+#include <stdint.h>
+#endif
+
+// Forward declarations of DTL contexts for the underlying implementations
+struct dyad_dtl_ucx;
+struct dyad_dtl_flux;
+
+// Union type to store the underlying DTL contexts
+union dyad_dtl_private {
+    struct dyad_dtl_ucx* ucx_dtl_handle;
+    struct dyad_dtl_flux* flux_dtl_handle;
+};
+typedef union dyad_dtl_private dyad_dtl_private_t;
+
+enum dyad_dtl_comm_mode {
+    DYAD_COMM_NONE = 0, // Sanity check value for when connection isn't established
+    DYAD_COMM_RECV = 1, // DTL connection will only receive data
+    DYAD_COMM_SEND = 2, // DTL connection will only send data
```

Comment:
add DYAD_COMM_END = 3 and no comma at the end

Reply:
Discussed over Slack and added